src/main/scala/scala/concurrent/stm/skel/TxnHashTrie.scala

```diff
@@ -201,12 +202,12 @@ private[skel] object TxnHashTrie {
 
       // class manifests for classes that have type parameters are a bit
       // convoluted to construct, so it is better to do it only once per split
-      implicit val cm = implicitly[ClassManifest[Node[A, B]]]
```

Comment:
This was a tricky line, because adding the value type creates an infinite loop (somehow it "works" without). Therefore, pass `cm` explicitly below.